src/driver/mysql/MysqlQueryRunner.ts

```diff
@@ -99,12 +100,15 @@ export class MysqlQueryRunner extends BaseQueryRunner implements QueryRunner {
     /**
      * Starts transaction on the current connection.
      */
-    async startTransaction(): Promise<void> {
+    async startTransaction(isolationLevel?: IsolationLevel): Promise<void> {
         if (this.isTransactionActive)
             throw new TransactionAlreadyStartedError();
 
         this.isTransactionActive = true;
         await this.query("START TRANSACTION");
+        if (isolationLevel) {
+            await this.query("SET SESSION TRANSACTION ISOLATION LEVEL " + isolationLevel);
```

Comment:
Wouldn't it be better to use `"SET TRANSACTION ISOLATION LEVEL " + isolationLevel` (without SESSION) _before_ `"START TRANSACTION"`, so isolation level will only apply to one transaction, not all subsequent transactions within the current session?

Reply:
You are correct, this was an oversight on my part.  I'll make the change and push up another PR.

Reply:
Actually, I'll have to change the logic so that the `SET` happens before the transaction starts.